doc/port-config-refactor/port-config-refactor-design.md

```diff
@@ -0,0 +1,48 @@
+# SONiC Port Configuration Refactor Design #
+
+## Rev 0.1 ###
+
+### Revision ###
+
+ | Rev |     Date    |       Author       | Change Description                |
+ |:---:|:-----------:|:------------------:|-----------------------------------|
+ | 0.1 |             |      Junchao Chen  | Initial version                   |
+
+## 1. Overview
+
+The file "port_config.ini" defines port name, index and other information. In current SONiC code, there is a module [portconfig.py](https://github.com/Azure/sonic-buildimage/blob/master/src/sonic-config-engine/portconfig.py) to parse and collect all information in this file. However, there are also other places have code to parse "port_config.ini". To keep code clean, it would be better to make all those parse logic reuse [portconfig.py](https://github.com/Azure/sonic-buildimage/blob/master/src/sonic-config-engine/portconfig.py).
+
+There are also open PRs to begin transitioning from the "port_config.ini" file to a new "platform.json" file. So if we keep all parse logic in [portconfig.py](https://github.com/Azure/sonic-buildimage/blob/master/src/sonic-config-engine/portconfig.py), it will be easy to keep backward compatible.
```

Comment:
I don't understand this part. How does this help with backwards compatibility?

Reply:
Now we have multiple places to parse port_config.ini file, if we are going to replace port_config.ini file with platform.json, we need change all that places. However, if we keep the parse logic in portconfig.py, we only need change this place.